docs/usage.rst

```diff
@@ -36,7 +36,9 @@ Once enabled, *sphinx-click* enables automatic documentation for
    The following options are optional:
 
    ``:show-nested:``
-     Enable full documentation for sub-commands.
+     Set to "full" to enable full documentation for sub-commands. Set to "no"
+     to entirely omit sub-commands. Set to short to only list sub-commands.
+     Defaults to "short".
```

Comment:
I think this is going to be seen as rather confusing, since say `short` is the default, yet if you specify `:show-nested:` without arguments (which would normally mean "use defaults" in other contexts), you get the `full` behaviour. I'm tempted to say we consider this deprecated and add a new parameter, though I can't think of a good name yet.

Regardless of the option name chosen, can we use `none` instead of `no` and display this using a definition list, e.g.

    Configure documentation for sub-commands. Accepts three values:

    ``short`` (default)
      List sub-commands without documentation.

    ``full``
      List sub-commands with full nested documentation.

    ``none``
      Do not list sub-commands.

Reply:
Ah, you're right that it's ambiguous. I mostly focused on guaranteeing the same results for existing users (e.g.: backwards-compat above all).

Reply:
Agree with focusing on backwards compatibility. A new option that is mutually exclusive with `:show-nested:`, plus a note in the docs suggesting `:show-nested:` is deprecated (though not for removal, since it costs almost nothing to keep around) should do the trick. If you're happy to work on this, I'll leave naming of the option to you and simply critique from afar :smile:

Reply:
Added a new param :)